include/entity/triangularMesh.h

```diff
@@ -0,0 +1,27 @@
+#ifndef ENTITY_TRIANGLEMESH_H
+#define ENTITY_TRIANGLEMESH_H
+
+#include <array>
+#include <vector>  
+#include <graphic/color.h>
+#include <geometry/surfaceIntersection.h>
+#include <geometry/ray.h>
+#include <entity/box.h>  
+
+class TriangularMesh : public Box {
+private:
+    int numTriangles;
+    int numVertices;
+    std::vector<Point> vertices;
+    std::vector<Vetor> normals;
+    std::vector<std::array<int, 3>> triangleIndices;
+    std::vector<Vetor> triangleNormals;
+    std::vector<Vetor> vertexNormals;
+    std::vector<Color> colors; 
+
+public:
+    TriangularMesh();
+    TriangularMesh(double kd, double ks, double ka, double kr, double kt, double roughness);
```

Comment:
```suggestion
    TriangularMesh(
        int numberOfTriangles,
        int numberOfVertices,
        std::vector<Point> vertices,
        std::vector<std::array<int, 3>> triangles,
        std::vector<Vetor> triangleNormals,
        std::vector<Vetor> vertexNormals,
        Color color,
        double kd,
        double ks,
        double ka,
        double kr,
        double kt,
        double roughness
    );
```
Acho válido ter um constructor completo, que já inicialize tudo e não só os atributos relacionados ao `Box`

Reply:
também acho ter, mas acho que agora os construtores estão ficando cada vez mais complexos e verbosos, sendo a ideia de evoluir isso pra uma API, acho que isso fica bem ruim. Precisamos discutir uma forma de limpar esses construtores.

Reply:
é pq o projeto tá sem linter. Se tivesse ficaria mais fácil de verificar isso. Um método não pode ter muitos parâmetros e nem ser muito extenso. Mas isso não é algo bloqueante por agora, como a  gente ainda precisa decidir, a gente pode consertar essas coisas do projeto num refactor futuro.

Reply:
A gente pode adicionar getters/setters, que nem fizemos no `Scene`, para deixar tudo mais dinâmico